src/cfnlint/__init__.py

```diff
@@ -758,39 +765,42 @@ def check_value(self, obj, key, path,
         for value_obj in values_obj:
             value = value_obj['Value']
             child_path = value_obj['Path']
-            if not child_path:
-                if check_value:
-                    matches.extend(
-                        check_value(
-                            value=value, path=new_path[:] + child_path, **kwargs))
-            elif child_path[-1] == 'Ref':
-                if check_ref:
-                    matches.extend(
-                        check_ref(
-                            value=value, path=new_path[:] + child_path,
-                            parameters=self.get_parameters(),
-                            resources=self.get_resources(),
-                            **kwargs))
-            elif child_path[-1] == 'Fn::FindInMap':
-                if check_mapping:
-                    matches.extend(
-                        check_mapping(
-                            value=value, path=new_path[:] + child_path, **kwargs))
-            elif child_path[-1] == 'Fn::Join':
-                if check_join:
-                    matches.extend(
-                        check_join(
-                            value=value, path=new_path[:] + child_path, **kwargs))
-            elif child_path[-1] == 'Fn::Split':
-                if check_split:
-                    matches.extend(
-                        check_split(
-                            value=value, path=new_path[:] + child_path, **kwargs))
-            elif isinstance(child_path[-1], int):
+            if not isinstance(value, dict):
                 if check_value:
                     matches.extend(
                         check_value(
                             value=value, path=new_path[:] + child_path, **kwargs))
+            else:
+                if len(value) == 1:
+                    for dict_name, _ in value.items():
+                        if dict_name in cfnlint.helpers.FUNCTIONS:
+                            function_name = 'check_%s' % camel_to_snake(dict_name.replace('Fn::', ''))
```

Comment:
Ugh.  We already discussed this.  Still, ugh. :)
Could you maybe throw a comment in here for when I inevitably forget what we discussed?